nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-api/src/main/java/org/nuxeo/ecm/directory/BaseSession.java

```diff
@@ -443,6 +443,20 @@ public void deleteEntry(String id) throws DirectoryException {
         getDirectory().invalidateCaches();
     }
 
+    /**
+     * @param results the query results without limit and offet
+     * @param limit maximum number of results ignored if less than 1
+     * @param offset number of rows skipped before starting, will be 0 if less than 0.
+     * @return the result with applied liimit and offset
+     * @since 10.1
+     * @see {@link Session#query(Map, Set, Map, boolean, int, int)}
+     */
+    protected DocumentModelList applyQueryLimits(DocumentModelList results, int limit, int offset) {
+        int safeOffset = Math.max(0, offset);
+        int toIndex = limit >= 1 ? Math.min(results.size(), safeOffset + limit) : results.size();
```

Comment:
Use `>= 0` instead. There's no reason to make a limit of `0` return the whole list, it's a valid degenerate case.
And of course make sure there's tests for this case.

Reply:
Yes it's unfortunate. Keep it then.